java/client/src/main/java/glide/api/models/commands/SetOptions.java

```diff
@@ -0,0 +1,138 @@
+package glide.api.models.commands;
+
+import glide.api.models.exceptions.RequestException;
+import java.util.LinkedList;
+import java.util.List;
+import lombok.Builder;
+import lombok.NonNull;
+
+/**
+ * Object builder to add optional arguments to {@link glide.api.commands.StringCommands#set(String,
+ * String, SetOptions)}
+ */
+@Builder
+public class SetOptions {
+
+    /**
+     * If `conditional` is not set the value will be set regardless of prior value existence. <br>
+     * If value isn't set because of the condition, return null.
+     */
+    private ConditionalSet conditionalSet;
+
+    /**
+     * Return the old string stored at key, or null if key did not exist. An error is returned and SET
+     * aborted if the value stored at key is not a string. Equivalent to `GET` in the Redis API.
+     */
+    private boolean returnOldValue;
+
+    /** If not set, no expiry time will be set for the value. */
+    private TimeToLive expiry;
+
+    public enum ConditionalSet {
+        /**
+         * Only set the key if it does not already exist. <br>
+         * Equivalent to `NX` in the Redis API.
+         */
+        ONLY_IF_EXISTS,
+        /**
+         * Only set the key if it already exists. <br>
+         * Equivalent to `EX` in the Redis API.
+         */
+        ONLY_IF_DOES_NOT_EXIST
+    }
+
+    @Builder
+    public static class TimeToLive {
+        /** Expiry type for the time to live */
+        @NonNull private TimeToLiveType type;
+
+        /**
+         * The amount of time to live before the key expires. Ignored when KEEP_EXISTING type is set.
+         */
+        private Integer count;
+    }
+
+    public enum TimeToLiveType {
+        /**
+         * Retain the time to live associated with the key. <br>
+         * Equivalent to `KEEPTTL` in the Redis API.
```

Comment:
Maybe add hyperlinks? (below too)
```suggestion
         * Equivalent to <a href="....">KEEPTTL</a> in the Redis API.
```
or at least
```suggestion
         * Equivalent to <em>KEEPTTL</em> in the Redis API.
```